### PR TITLE
Show Current Location functionality for the web

### PR DESCRIPTION
**Pull request by vkkostin**
The `Show Current Location` toggle is currently not supported for web apps:
Screen Shot 2022-04-25 at 8 52 08 AM

This PR adds that support:
Screen Shot 2022-04-25 at 8 52 36 AM

The most challenging part of this was loading in current location independently of other data - asking for browser location permissions, loading in the current location, and displaying it on the map should _not_ block other addresses being loaded in and displayed on the map. If the user denies location permissions, the map should still display other addresses and be fully functional. To do this I made the current location logic be parallel (but independent) of the marker collection logic. They both have their own loading logic, functions that check whether or not to "unload" the map component with new data, and functions that transform the data into something the Google map component expects.

### CODE REVIEW

**Comment by vkkostin** [comment]
[Maps Component: Current Location Feature Doesn't Work on Web](https://trello.com/c/6qgsDCKx/144-maps-component-current-location-feature-doesnt-work-on-web)

File: src/Map/index.js [hunk 1]
```diff
@@ -184,14 +254,19 @@ export default class Map extends Component {
...
-      addresses: geocodedCoordinates,
-      isLoaded: true,
-      isLoading: false,
+      dataAddresses: geocodedCoordinates,
+      isDataAddressesLoaded: true,
+      isDataAddressesLoading: false,
     })
   }
 
-  getFilteredAddresses = () => {
```
Comment (adebroux) [inline src/Map/index.js:193]: There's a comment at line 249 that refers to this `getFilteredAddresses` function, so if we're renaming this function we should probably update that comment.

File: src/Map/index.js [hunk 2]
```diff
@@ -84,54 +88,120 @@ export default class Map extends Component {
...
   }
 
-  mapShouldReload() {
+  /**
+   * WEB ONLY
+   * Tells the component whether or not to reload the map based on if the browser's
+   * current position is in the final array of addresses that are given to the map
+   * @returns {boolean}
+   */
+  shouldUpdatedUserAddress() {
```
Comment (adebroux) [inline src/Map/index.js:140]: Nitpick: should this be `shouldUpdateUserAddress`?

Comment (adebroux) [inline src/Map/index.js:140]: Nitpick: should this be `shouldUpdateUserAddress`?

**Review by adebroux — CHANGES_REQUESTED** [review]
Looks great! Just some minor comments about function names

**Review by adebroux — APPROVED** [review]
LGTM!

**Pull request merged**